### ISSUE TITLE
Move Curseforge Blocked mods automatically into mods folder

### ISSUE THREAD

**Issue by TayouVR** [PolyMC/PolyMC#918]
### Role

I am using curseforge modpacks

### Suggestion

When downloading blocked mods manually listen on the downloads folder and automatically move mods to the appropriate mods folder if possible. ATLauncher has a similar feature.

### Benefit

It would make using curseforge packs easier

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

You may find inspiration for how to implement such a feature here: https://github.com/ATLauncher/ATLauncher/blob/0035b39da1e0c9a79e0b2b54858e308701554f84/src/main/java/com/atlauncher/gui/dialogs/BrowserDownloadDialog.java#L123

Also this would go hand-in-hand with #820

**Comment by DDriggs00** [comment]
This would additionally need support for those with non-standard browser download locations

**Comment by TayouVR** [comment]
> This would additionally need support for those with non-standard browser download locations

yes, maybe there are functions to figure out the download folder in Qt, if not it might require a extra lib or some helper function to figure out where downloads are on all respective platforms

**Comment by Ricky12Awesome** [comment]
it should be possible for it to open a browser with a custom profile with the download directory to a specified path

Chromium based would be `chromium-browser --profile-directory="Profile X" --new window link/to/mod`

Firefox based would be `firefox --profile <path> --new-window link/to/mod`

**Comment by TayouVR** [comment]
hmm, that is a interesting idea... The download path could be set directly to the mods folder of the instance in question.
However this would not work in sandboxes like flatpak.. at least not if the browser is installed through flatpak. (as it wouldn't have file access to the custom profile)

Also it would mean, that PolyMC would have to figure out which browser is installed (and preferred) and build a profile for that browser/have one packaged with it
This is probably not really viable...

**Comment by Ricky12Awesome** [comment]
Actually I think doing `--user-data-dir=/path/to/dir` would probably be better, since you should be able to use the same directory for any chromium based browsers

And for detecting browsers, just grab the default browser *(or from there settings if they want to use a different browser)*, check if it's firefox or safari, if it's not one of those then just assume it's a chromium based browser, if that somehow fails, then fallback to just simply opening in the default browser

It's unlikely for it to fail since like over 99% of users will use at least one of those browsers (usually just chrome)

**Comment by TheDeviantCrafter** [comment]
ATLauncher used something similar back when mods were still behind Adfly links. They just added a setting to locate the downloads folder if it wasn't in libraries.

**Comment by TayouVR** [comment]
oh yeah, I guess the simplest option would be to have a Downloads folder setting similar to instances, etc. 
it just defaults to `~/Downloads` or `%UserDir%\Downloads`, for which there might be a Qt variable/function too, which would make it easier. (e.g. folder name could be localized)

That way we don't need a complex detection code or anything.

**Comment by TheDeviantCrafter** [comment]
Anyone with a non-standard downloads location is probably tech savvy enough to change the path in settings.

PolyMC could check if the downloads folder exists at the default location and warn the user if it doesn't. Add some red text to the open download link dialog.

**Comment by TechPro424** [comment]
So, as far as I can understand from the above discussion, the current method is to open the user's default browser, download the mod, and put it in the mods folder, right?

**Comment by TayouVR** [comment]
Yes, PolyMC currently doesn't move the mod from the downloads folder to the mod folder automatically, but that is what he hope to achieve at some point. 
I have shown interest on working a PR for this myself, but my lack of knowledge in C++ makes it a bit difficult. If anyone knows C++ better and wants to do this, you are welcome!

**Comment by DovydasTEDS** [comment]
I would love this feature!

**Comment by Maple38** [comment]
> This would additionally need support for those with non-standard browser download locations

Couldn't there just be an option in settings to change the download location? The majority of people likely have it in the default downloads folder.

**Comment by TayouVR** [comment]
> Couldn't there just be an option in settings to change the download location? The majority of people likely have it in the default downloads folder.

yes I think that would make sense
